Combine `set` actions

CoSNaYe opened the pull request:
1. Add "Combine `set` actions" rule
2. Add `Under consideration` and `References` sections
3. add rule: Prefer single quote or backtick for String.  …
4. add rule: Add `propTypes` to every components
5. add rule: Use multiple lines when we need to pass more than one props.

Commit:
add some setMutations

foodaka commented on README.md:
```diff
@@ -51,6 +64,12 @@ A coding guide line for React + Redux project @WORD UP
   * [Normalizing State Shape](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape)
 
+#### Combine `set` actions
+  * Instead of using `state.set('key1', value1).set('key2', 'value2')`, use `withMutations` to wrap multiple actions into one update. Since Immutable.js will perform re-arranging in all the processes and retain all intermediate states.
```
I think we should give an actual example here of what it should look like

CoSNaYe replied:
good idea, added in 1ffe8dc

foodaka commented on README.md:
```diff
@@ -51,6 +64,12 @@ A coding guide line for React + Redux project @WORD UP
+#### Combine `set` actions
+  * Instead of using `state.set('key1', value1).set('key2', 'value2')`, use `withMutations` to wrap multiple actions into one update. Since Immutable.js will perform re-arranging in all the processes and retain all intermediate states.
+
+#### Refrain from using `.toJS()`
```
good

CoSNaYe committed:
1. add rule: Prefer single quote or backtick for String.
2. add example
3. add rule: Add `propTypes` to every components
4. add rule: Use multiple lines when we need to pass more than one props.

CoSNaYe committed:
add BEM style for CSS

CoSNaYe committed:
minor changes

CoSNaYe committed:
minor

CoSNaYe commented:
Slack integration test.
